### PR TITLE
feat(errors): make verrors=context global and update test outputs

### PR DESCRIPTION
**Pull request by royalpinto007**
Make `verror=context` usage global and update related test files accordingly.

### CODE REVIEW

File: compiler/src/dmd/mars.d [hunk 1]
```diff
@@ -631,6 +631,15 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
...
         }
 
+        if (arg == "-verrors=context")
+        {
+            global.params.v.printErrorContext = true; // Ensure this is enabled
+        }
+        else if (arg == "-verrors=none")
+        {
+            global.params.v.printErrorContext = false; // Ensure this is disabled
+        }
```
Comment (thewilsonator) [inline compiler/src/dmd/mars.d:641]: this change should be made https://github.com/dlang/dmd/blob/master/compiler/src/dmd/mars.d#L1010 and not here

Comment (royalpinto007) [reply compiler/src/dmd/mars.d]: done

File: compiler/src/dmd/mars.d [hunk 2]
```diff
@@ -631,6 +631,11 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
             continue;
         }
 
+        if (arg == "-verrors=context")
+        {
+            global.params.v.printErrorContext = true; // Ensure this is enabled
+        }
```
Comment (thewilsonator) [inline compiler/src/dmd/mars.d:637]: and this check is already done...

File: compiler/src/dmd/globals.d [hunk 3]
```diff
@@ -170,7 +170,6 @@ extern (C++) struct Param
     bool addMain;           // add a default main() function
     bool allInst;           // generate code for all template instantiations
     bool bitfields;         // support C style bit fields
-
```
Comment (thewilsonator) [inline compiler/src/dmd/globals.d:173]: this is not needed

**Comment by dkorpel** [comment]
Was it decided somewhere that this should become the default? I prefer the current default, and don't like how this makes the test suite harder to maintain.

**Comment by maxhaton** [comment]
> Was it decided somewhere that this should become the default? I prefer the current default, and don't like how this makes the test suite harder to maintain.

Don't think so, but we are one of not many languages still not doing the squiggle by default. That debate aside, the last time I think someone (hatf0 I think) tried to turn it a bunch of things blew up so I asked @royalpinto007 to look into just to see how things were going on that front?

**Comment by bangbangsheshotmedown** [comment]
> Was it decided somewhere that this should become the default? I prefer the current default, and don't like how this makes the test suite harder to maintain.

I like this PR

With the code example, i know exactly what's wrong without having to decipher a lengthily message

I use Zig a lot and this is so much better

**Comment by royalpinto007** [comment]
I need some help with the `compilable/testcolor.sh` file. I tried resolving it, but unfortunately, I wasn't able to.

**Comment by thewilsonator** [comment]
ping @maxhaton please review

**Review by RazvanN7 — APPROVED** [review]
I like it.

File: changelog/dmd.verror_context.dd [hunk 4]
```diff
@@ -0,0 +1,33 @@
+Make `verror=context` Usage Global and Update Related Test Files
```
Comment (dkorpel) [inline changelog/dmd.verror_context.dd:1]:
```suggestion
Enable `-verrors=context` by default
```

Test suite changes are irrelevant to users.

File: changelog/dmd.verror_context.dd [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+The `verror=context` error handling mechanism has been made global, ensuring that error messages now consistently include context information across the entire codebase. Previously, error handling was localized, leading to some errors lacking context in the output.
```
Comment (dkorpel) [inline changelog/dmd.verror_context.dd:3]: This is a weird description: all it does is print the line of code the error applies to under the message. It has nothing to do with 'across the entire codebase' vs 'localized'.

**Review by dkorpel — CHANGES_REQUESTED** [review]
I don't know if it's a good idea to switch the default, I'll bring that up tomorrow in the DLF monthly meeting.

However, the test suite doesn't have to use the default. It's interesting to see what all these errors now look like with -verrors=context, but it also makes me notice that often, the column location is wrong. So this is very useful to spot all the places where error locations can be improved, but perhaps we shouldn't promote -verrors=context yet while it's inaccurate so many times.

I also notice the way the tests are updated is that `#line` markers are removed and all test output blocks are merged and put at the beginning. This means that now, a small change in the test will cause a large diff in a big output block. The original structure was supposed to prevent that. I have been working on a special test runner directive that allows local, line number-independent error messages: https://github.com/dlang/dmd/discussions/14515 

In any case, I don't think the 5000 new copied lines of code with an error are worth adding to the test suite.

**Comment by maxhaton** [comment]
Consensus: enable this for users, disable it in almost all of the test suite for visual noise.

@royalpinto007 thank you very much for updating the tests though. We can't enable it without making sure it works acorss the whole test suite without blowing up! Now we can.

Can we name the option to turn it off `-verrors=simple`?

**Pull request closed without merging**